x-pack/filebeat/input/httpjson/internal/v2/input.go

```diff
@@ -0,0 +1,225 @@
+// Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+// or more contributor license agreements. Licensed under the Elastic License;
+// you may not use this file except in compliance with the Elastic License.
+
+package v2
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"net"
+	"net/http"
+	"net/url"
+	"time"
+
+	retryablehttp "github.com/hashicorp/go-retryablehttp"
+	"go.uber.org/zap"
+
+	v2 "github.com/elastic/beats/v7/filebeat/input/v2"
+	inputcursor "github.com/elastic/beats/v7/filebeat/input/v2/input-cursor"
+	"github.com/elastic/beats/v7/libbeat/beat"
+	"github.com/elastic/beats/v7/libbeat/common"
+	"github.com/elastic/beats/v7/libbeat/common/transport/tlscommon"
+	"github.com/elastic/beats/v7/libbeat/common/useragent"
+	"github.com/elastic/beats/v7/libbeat/logp"
+	"github.com/elastic/go-concert/ctxtool"
+	"github.com/elastic/go-concert/timed"
+)
+
+const (
+	inputName = "httpjson"
+)
+
+var (
+	userAgent = useragent.UserAgent("Filebeat")
+
+	// for testing
+	timeNow = time.Now
+)
+
+type retryLogger struct {
+	log *logp.Logger
+}
+
+func newRetryLogger(log *logp.Logger) *retryLogger {
+	return &retryLogger{
+		log: log.Named("retryablehttp").WithOptions(zap.AddCallerSkip(1)),
+	}
+}
+
+func (log *retryLogger) Error(format string, args ...interface{}) {
+	log.log.Errorf(format, args...)
+}
+
+func (log *retryLogger) Info(format string, args ...interface{}) {
+	log.log.Infof(format, args...)
+}
+
+func (log *retryLogger) Debug(format string, args ...interface{}) {
+	log.log.Debugf(format, args...)
+}
+
+func (log *retryLogger) Warn(format string, args ...interface{}) {
+	log.log.Warnf(format, args...)
+}
+
+func newTLSConfig(config config) (*tlscommon.TLSConfig, error) {
+	if err := config.Validate(); err != nil {
+		return nil, err
+	}
+
+	tlsConfig, err := tlscommon.LoadTLSConfig(config.Request.SSL)
+	if err != nil {
+		return nil, err
+	}
+
+	return tlsConfig, nil
+}
+
+func test(url *url.URL) error {
+	port := func() string {
+		if url.Port() != "" {
+			return url.Port()
+		}
+		switch url.Scheme {
+		case "https":
+			return "443"
+		}
+		return "80"
+	}()
+
+	_, err := net.DialTimeout("tcp", net.JoinHostPort(url.Hostname(), port), time.Second)
+	if err != nil {
+		return fmt.Errorf("url %q is unreachable", url)
+	}
+
+	return nil
+}
+
+func run(
+	ctx v2.Context,
+	config config,
+	tlsConfig *tlscommon.TLSConfig,
+	publisher inputcursor.Publisher,
+	cursor *inputcursor.Cursor,
+) error {
+	log := ctx.Logger.With("url", config.Request.URL)
+
+	stdCtx := ctxtool.FromCanceller(ctx.Cancelation)
```

Comment:
```suggestion
	stdCtx := ctxtool.FromCanceller(ctx.Cancellation)
```

It's outside the scope of this PR, but thought I'd note this is spelled wrong.